addons/mail/static/src/components/welcome_view/welcome_view.js

```diff
@@ -0,0 +1,34 @@
+/** @odoo-module **/
+
+import { registerMessagingComponent } from '@mail/utils/messaging_component';
+import { useRefToModel } from '@mail/component_hooks/use_ref_to_model/use_ref_to_model';
+import { useUpdateToModel } from '@mail/component_hooks/use_update_to_model/use_update_to_model';
+
+export class WelcomeView extends owl.Component {
+
+    /**
+     * @override
+     */
+    setup() {
+        super.setup();
+        useRefToModel({ fieldName: 'guestNameInputRef', modelName: 'mail.welcome_view', propNameAsRecordLocalId: 'localId', refName: 'guestNameInput' });
+        useUpdateToModel({ methodName: 'onComponentUpdate', modelName: 'mail.welcome_view', propNameAsRecordLocalId: 'localId' });
+    }
+
+    /**
+     * @returns {mail.welcome_view}
+     */
+    get welcomeView() {
+        return this.messaging.models['mail.welcome_view'].get(this.props.localId);
+    }
+}
+
+Object.assign(WelcomeView, {
+    props: {
+        localId: String,
+        mediaPreviewLocalId: String,
```

Comment:
As mentioned before, this should be a relation in `welcome_view` model, not an extra props.